### DISCUSSION TITLE
How to add this site to search engine ?

### DISCUSSION THREAD

**Discussion by Mitia99** [Pitmairen/selection-search#144] [Q&A] [answered]
Hi everyone,

Can anyone help?

I'm struggling trying to figure how to add [this site](https://www.french-stream.gg/) to search engine.

My issue is that the URL doesn't change after performing a search so I don't know where to put the `%`

Thanks,

**Comment by Pitmairen** [comment, marked as answer]
Looks like that site is using http POST requests for the search. That's why there is no search changes to the url.

Selection search has support for POST searches, but it requires a bit more setup.  

Unless you have disabled it in the settings, it should be possible to hold CTRL+ALT on the keyboard then left click in the search input on the site. This should open a popup which allows you to automatically add the search engine.

If this doesn't work you can do it manually using the following url:

`https://www.french-stream.gg/{POSTARGS}story=%s&do=search&subaction=search{POSTENCODING}UTF-8` 

Then in the options for the search engine (wrench icon), you need to check the "use post method" checkbox:

![image](https://github.com/Pitmairen/selection-search/assets/357032/8373ea57-2e0b-45c4-924c-b22b7ccda787)

**Reply by Mitia99** [reply]
Thanks a lot for the tip. Worked as intended.

**Comment by Mitia99** [comment]
@Pitmairen
Hi,

Can you help again  for [this other site](https://www.hdss.plus/) please ? Also using POST request.

I tried with `https://www.hdss.plus/{POSTARGS}story=%s&do=search&subaction=search{POSTENCODING}UTF-8` but not working.

I thought configuration will be similar to previous one.

Thanks,

**Reply by Pitmairen** [reply]
I tried the search url you posted, and it is working for me. Remember to check the "use post method" checkbox in the search engine options.

**Reply by Mitia99** [reply]
> I tried the search url you posted, and it is working for me. Remember to check the "use post method" checkbox in the search engine options. 

Retried and it works, maybe there was something wrong with the term I searched.

**Comment by Mitia99** [comment]
@Pitmairen How can I configure [this site](https://v5.voiranime.com/) if I want VOSTFR search results? It uses two search bar.
![Screenshot 2023-11-02 135552](https://github.com/Pitmairen/selection-search/assets/56766778/41d2601c-8c7b-4966-b950-4c5ac13f9a07)

Ty for letting me know.

**Reply by Pitmairen** [reply]
This should work I think:
```
https://v5.voiranime.com/?s=%s
```

**Reply by Mitia99** [reply]
> This should work I think:
> 
> ```
> https://v5.voiranime.com/?s=%s
> ```

It still displays VF search results.

**Reply by Pitmairen** [reply]
This should work then:
```
https://v5.voiranime.com/?s=%s&language=vostfr
```

**Reply by Mitia99** [reply]
> This should work then:
> 
> ```
> https://v5.voiranime.com/?s=%s&language=vostfr
> ```
Ty. This one worked.

**Comment by Mitia99** [comment]
Hi,

@Pitmairen 
Can you help with [Sci-Hub website](https://sci-hub.se/) ?

It's kinda specific as results should appear like this

![image](https://github.com/user-attachments/assets/05e6280e-ddfa-4fed-b475-7671a9a22809)

Here's an exemple of an article you can search "https://doi.org/10.1016/j.pneumo.2012.01.002"

Thank you,

**Reply by Pitmairen** [reply]
Try this:
```
https://sci-hub.se/{%(RAW)s}
```

**Reply by Mitia99** [reply]
> Try this:
> 
> ```
> https://sci-hub.se/{%(RAW)s}
> ```

Working as intended.

Thank you,

**Comment by Mitia99** [comment]
@Pitmairen 
Is it possible to add [ChatGPT](https://chatgpt.com/) as search engine ?

I mean the free version without login.

**Reply by Pitmairen** [reply]
Yes, it's possible by using a user script. Install a user script extension like  Violentmonkey or Tampermonkey.

Then install this user script:
```javascript
// ==UserScript==
// @name        New script
// @namespace
// @match       https://chatgpt.com/*
// @grant       none
// @version     1.0
// @author      -
// @description 10/25/2024, 9:06:36 PM
// ==/UserScript==

(function() {
    'use strict';

  function setQuery(){
    const urlParams = new URLSearchParams(window.location.search);

    const q = urlParams.get("q")

    if (q){

      const prompt = document.querySelector("prompt-textarea")
      prompt.textContent = urlParams.get("q")

      const sendButton = document.querySelector('[data-testid="send-button"]')
      sendButton.click()

    }
  }
  setQuery()
})();
```


Now you can use the following search url:

```
https://chatgpt.com/?q=%s
```

**Reply by Mitia99** [reply]
@Pitmairen Thanks for the continuous support. I managed to install the script and it worked as intented.

Surprisingly, `https://chatgpt.com/?q=%s` also works without even Tampermonkey installed. I tried to remove the extension and no issue.

**Reply by Pitmairen** [reply]
Yes, looks like it actually works without the script. That's even better, haha :)
